### PR TITLE
feat: Restore room creation

### LINKED ISSUE #193: [UX/UI] Restore room creation functionality
**Type**
Feature

**Description**
Title

**Motivation / use case**
Demo purpose

### PR DESCRIPTION
**Pull request by pmrukot**
**Summary**
This PR restores functionality of creating a room with custom categories. 
* Added a `:create` to room resource at `router.ex`
* Restored tests deleted in #144 
* Added `Create room` to a navbar
* Small refactor of a `Panel/View.elm` as an addition

**Related issues**
Closes #193

**Test plan**
Tests + manual

**Screenshots**
screen shot 2017-11-20 at 23 40 55

### CODE REVIEW

File: aion/lib/aion/ranking.ex [hunk 1]
```diff
@@ -11,26 +11,41 @@ defmodule Aion.Ranking do
     UserCategoryScore
   }
 
-  @type category_score_t :: %{category_id: integer, category_name: String.t, scores: (list user_score_t)}
```
Comment (mrapacz) [inline aion/lib/aion/ranking.ex]: Not sure if it's a good idea to format the files that you don't touch. 

I'm 100% for running `mix format` on the ones that you edit anyway. As far as I remember the syntax is just `mix format filepath/folderpath`. Or you can `mix format` and then checkout the files that you didn't touch.

File: aion/test/controllers/room_controller_test.exs [hunk 2]
```diff
@@ -1,10 +1,11 @@
 defmodule Aion.RoomControllerTest do
   use Aion.AuthConnCase
 
-  alias Aion.{Room, Question, User}
+  alias Aion.{Room, Question, User, Category, RoomCategory}
   alias Aion.RoomChannel.{Monitor, QuestionSet}
 
   @room %Room{description: "Here come dat boi", name: "Dat Boi"}
+  @invalid_room %Room{name: "This is an invalid data, has no description field"}
```
Comment (mrapacz) [inline aion/test/controllers/room_controller_test.exs]: `The provided data is invalid, it does not contain the description field`
or 
`The provided data is invalid does not contain the description field`.

The data is singular uncountable, so there should be no `an` before it

**Review by mrapacz — COMMENTED** [review]
Half of the file diff is subject to `mix format` of files that were not edited. 

I think maybe they should have been checked out. Apart from that the code looks good.

File: aion/lib/aion/ranking.ex (hunk 1 above)
Comment (pmrukot) [reply aion/lib/aion/ranking.ex]: Yeah, did not notice while I was playing with formatter, will revert that.

**Review by mrapacz — APPROVED** [review]

**Pull request merged**